File: 백준/Bronze/13300. 방 배정/방 배정.java
import java.util.*;
import java.lang.*;
import java.io.*;

class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String s = br.readLine();
        StringTokenizer st = new StringTokenizer(s);
        int n = Integer.parseInt(st.nextToken());
        int k = Integer.parseInt(st.nextToken());

        int[][] info = new int[6][2];
        for(int i=0; i<n; i++) {
            s = br.readLine();
            st = new StringTokenizer(s);
            int gender = Integer.parseInt(st.nextToken());
            int grade = Integer.parseInt(st.nextToken());
            info[grade-1][gender]++;
        }

        int cnt = 0;
        for(int i=0; i<6; i++) {
            cnt += info[i][0]/k;
            if (info[i][0]%k > 0) cnt++;

            cnt += info[i][1]/k;
            if (info[i][1]%k > 0) cnt++;
        }
        System.out.println(cnt);
    }
}

File: 백준/Bronze/13300. 방 배정/방 배정.swift
import Foundation

let nk = readLine()!.split(separator: " ").map{Int(String($0))!}

// i+1학년 -> [여학생 수, 남학생 수]
var info = Array(repeating: Array(repeating: 0, count: 2), count: 6)
for _ in 0..<nk[0] {
    let stud = readLine()!.split(separator: " ").map{Int(String($0))!}
    if (stud[0] == 0) { info[stud[1]-1][0] += 1 } 
    else { info[stud[1]-1][1] += 1 }
}

var cnt = 0
for i in info {
    cnt += i[0]/nk[1]
    if (i[0]%nk[1] > 0) { cnt += 1 }
    
    cnt += i[1]/nk[1]
    if (i[1]%nk[1] > 0) { cnt += 1 }
}
print(cnt)